### PR TITLE
OS X notifications 

### PR DESCRIPTION
**Pull request by jendas1**
Here is first part of my job. Tested and working :)

### CODE REVIEW

File: src/plugins/backends/osxnotificationcenter/osxnotificationcenter.mm [hunk 1]
```diff
@@ -1,41 +1,114 @@
...
+#include "libsnore/log.h"
 
 #import <QThread.h>
 #import <QApplication.h>
+#import <QMap>
+#include <Foundation/Foundation.h>
 
 using namespace Snore;
 
+QMap<int, Notification> id_to_notification;
```
Comment (TheOneRing) [inline src/plugins/backends/osxnotificationcenter/osxnotificationcenter.mm]: I usually use m_camelCaseNames

File: src/plugins/backends/osxnotificationcenter/osxnotificationcenter.mm [hunk 2]
```diff
@@ -1,41 +1,114 @@
...
+{
+    
+    snoreDebug(SNORE_DEBUG) << "User clicked on notification";
+    int notification_id = [notification.userInfo[@"id"] intValue];
+    [center removeDeliveredNotification: notification];
+    if (not id_to_notification.contains(notification_id)) {
+        snoreDebug(SNORE_WARNING) << "User clicked on notification that was not recognized";
+        return;
+    }
+    Notification snore_notification = id_to_notification[notification_id];
```
Comment (TheOneRing) [inline src/plugins/backends/osxnotificationcenter/osxnotificationcenter.mm]: maybe id_to_notification.take() ?
I can't see that values are removed.

File: src/plugins/backends/osxnotificationcenter/osxnotificationcenter.mm [hunk 3]
```diff
@@ -1,41 +1,114 @@
...
 }
 
 void OSXNotificationCenter::slotNotify(Snore::Notification notification)
 {
-    NSUserNotification *osx_notification = [[NSUserNotification alloc] init];
-    osx_notification.title = NSStringFromQString(Utils::toPlainText(notification.title()));
-    osx_notification.informativeText = NSStringFromQString(Utils::toPlainText(notification.text()));
+    NSUserNotification * osx_notification = [[[NSUserNotification alloc] init] autorelease];
+    NSString * notification_id = [NSString stringWithFormat:@"%d",notification.id()];
+    osx_notification.title = NSStringFromQString(notification.title());
```
Comment (TheOneRing) [inline src/plugins/backends/osxnotificationcenter/osxnotificationcenter.mm]: does mac support any markup? <i> <b> etc?

File: src/plugins/backends/osxnotificationcenter/osxnotificationcenter.mm [hunk 4]
```diff
@@ -1,41 +1,114 @@
...
+    delegate = new UserNotificationItemClass();
+}
+
+OSXNotificationCenter::~OSXNotificationCenter()
 {
-    notification_center = [NSUserNotificationCenter defaultUserNotificationCenter];
+    delete delegate;
+}
 
+bool OSXNotificationCenter::initialize()
```
Comment (TheOneRing) [inline src/plugins/backends/osxnotificationcenter/osxnotificationcenter.mm]: can be removed as its identical to SnoreBackend::initialize()

**Comment by jendas1** [comment]
I've updated the code to CamelCase. The OS X notification center doesn't support any formatting so I couldn't add the markup.

**Comment by TheOneRing** [comment]
Cool big thanks

**Pull request merged**